### PR TITLE
Adds MariaDB and connection

### PR DESCRIPTION
**Pull request by Linkinlog**
## What is getting changed with this PR
- Adds new mariadb container and databases
- Adds Go connection and test case
- Adds Make commands 
- Adds documentation

## Why are we making these changes?
- So we can have a database

## Outline testing instructions
- Checkout this branch
- Review the updated README.md in the files changed
- Duplicate env files and change the two that say "REPLACEME" to your preference
- (Optional) Use a database application and test connecting to the database

## Commands to run
```BASH
make rebuild
make test
```

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] If it is a large change, I have added thorough unit tests if applicable.
- [x] The testing instructions accurately explain how to confirm that this code change works
- [x] I have requested a review from at least 1 other person

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -25,6 +25,10 @@ FROM ${NODE_IMAGE}:${NODE_VERSION} as hb_front
 WORKDIR /usr/src/app
 
 COPY docker-entrypoint.sh /
-RUN chmod u+x /docker-entrypoint.sh
+
+COPY ["frontend/package.json", "frontend/package-lock.json", "/usr/src/app/"]
```
Comment (mitrerogelio) [inline Dockerfile:29]: dalton moment

File: docker-compose.yml [hunk 2]
```diff
@@ -1,22 +1,49 @@
...
     volumes:
       - ./backend:/usr/src/app
+      - ./docker/server/docker-entrypoint-init.d:/docker-entrypoint-init.d
     ports:
       - 8085:8080
-    command: go
+    command: server
+    depends_on:
+      - db
+
```
Comment (mitrerogelio) [inline docker-compose.yml:19]: Any particular reason for the empty line?

**Review by mitrerogelio — APPROVED** [review]
Chefkiss*

File: docker-compose.yml (hunk 2 above)
Comment (Linkinlog) [reply docker-compose.yml:19]: Just for spacing

**Review by mitrerogelio — CHANGES_REQUESTED** [review]
Remove Maria from `backend/.env.example`

**Review by mitrerogelio — APPROVED** [review]
wonderful

**Pull request merged**